core/libs/test_utils/src/assert.rs

```diff
@@ -206,7 +207,12 @@ pub fn server_work_paths(db: &Core, expected_paths: &[&'static str]) {
             let context = db.context(tx).unwrap();
             let account = context.tx.account.get(&OneKey {}).unwrap();
             let remote_changes = context.get_updates().unwrap().file_metadata;
-            let mut remote = context.tx.base_metadata.stage(remote_changes).to_lazy();
+            let mut remote = context
+                .tx
+                .base_metadata
+                .stage(remote_changes)
+                .pruned()
```

Comment:
remote changes must now be pruned because `stage` doesn't automatically prune (this logic slightly duplicates `calculate_work` logic)